src/appengine/handlers/base_handler.py

```diff
@@ -262,6 +264,22 @@ def redirect(self, url, **kwargs):
     check_redirect_url(url)
     super(Handler, self).redirect(url, **kwargs)
 
+  # TODO(mbarbella): Delete this once the Python 3 migration is complete.
```

Comment:
This is the main change from the previous version. It post-processes all headers to make sure we won't leave any as newstr.